projects/packages/image-cdn/src/class-image-cdn.php

```diff
@@ -0,0 +1,1386 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+/**
+ * Class for photon functionality.
+ *
+ * @package automattic/jetpack-image-cdn
+ */
+
+namespace Automattic\Jetpack\Image_CDN;
+
+/**
+ * Class Image_CDN
+ */
+class Image_CDN {
+
+	const PACKAGE_VERSION = '0.1.0-alpha';
+
+	/**
+	 * Singleton.
+	 *
+	 * @var null
+	 */
+	private static $instance = null;
+
+	/**
+	 * Allowed extensions.
+	 *
+	 * @var string[] Allowed extensions must match https://code.trac.wordpress.org/browser/photon/index.php#L41
+	 */
+	protected static $extensions = array(
+		'gif',
+		'jpg',
+		'jpeg',
+		'png',
+		'webp', // Jetpack assumes Photon_OpenCV backend class is being used on the server. See link in docblock.
+	);
+
+	/**
+	 * Image sizes.
+	 *
+	 * Don't access this directly. Instead, use self::image_sizes() so it's actually populated with something.
+	 *
+	 * @var array Image sizes.
+	 */
+	protected static $image_sizes = null;
+
+	/**
+	 * Singleton implementation
+	 *
+	 * @return object
+	 */
+	public static function instance() {
+		if ( ! is_a( self::$instance, 'Automattic\Jetpack\Image_CDN' ) ) {
+			self::$instance = new Image_CDN();
+			self::$instance->setup();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Silence is golden.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Register actions and filters, but only if basic Photon functions are available.
+	 * The basic functions are found in ./functions.photon.php.
+	 *
+	 * @uses add_action, add_filter
+	 * @return void
+	 */
+	private function setup() {
```

Comment:
When this package is present, it will prevent the photon module from loading to avoid any potential conflict. The module will still show enabled in the UI, but this package will be active under the hood.

Reply:
> When this package is present

Well, when Boost is present. See my reply to https://github.com/Automattic/jetpack/pull/29561#discussion_r1151620450.

Reply:
Nice; I see @haqadn has moved the compatibility checks into the module in 3835de43d83ec79200ded9a9f02e3582800eb041.